src/services/user.ts

```diff
@@ -238,8 +243,14 @@ const getAdvisors = (username: string): Promise<StudentAdvisorInfo[]> =>
 
 const getMailboxCombination = () => http.get('profiles/mailbox-combination/');
 
+const getBuildings = (): Promise<string[]> => http.get(`advancedsearch/buildings`);
+
 const setMobilePhoneNumber = (value: number) => http.put(`profiles/mobile_phone_number/${value}/`);
 
+const updateOfficeLocation = (OfficeLocation: OfficeLocationQuery) => {
+  return http.patch(`profiles/office_location`, OfficeLocation);
+};
```

Comment:
2 things:

1) I'm pretty sure your route is an `async` route
2) if so, you'll either need to add a `Promise< [YOUR OBJECT NAME] >` wrapper 

OR 

Just don't return anything, but then you might not know if the update succeeded 🤷🏼

Reply:
They're already returning a promise, it's just an inferred type.

Reply:
hmmm valid, but is it not better to define the return type?

Reply:
We based updateOfficeLocation from setMobilePhoneNumber which was also a inferred type that did not define the return type:
![image](https://github.com/gordon-cs/gordon-360-ui/assets/115328139/c71ea0eb-8987-4ccc-925f-db7e512bb2a0)

Reply:
We based updateOfficeLocation from setMobilePhoneNumber, which also returned a promise without defining the return type:
![image](https://github.com/gordon-cs/gordon-360-ui/assets/115328139/e49507a5-bc77-4d28-af96-f1f875bbb208)

Reply:
sounds good, if Evan is good with it then 👍🏼

Reply:
It definitely is better as a rule to define the return type, because currently it's being inferred as `Promise<unknown>`, so if the route is actually returning data than we should document that type. If it isn't actually returning data, you can just make it `Promise<undefined>`.